src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory.cs

```diff
@@ -615,7 +615,7 @@ private IObjectOrCollectionInitializerOperation CreateBoundCollectionInitializer
             return new LazyObjectOrCollectionInitializerExpression(initializers, _semanticModel, syntax, type, constantValue, isImplicit);
         }
 
-        private IOperation CreateBoundObjectInitializerMemberOperation(BoundObjectInitializerMember boundObjectInitializerMember)
+        private IOperation CreateBoundObjectInitializerMemberOperation(BoundObjectInitializerMember boundObjectInitializerMember, bool inCollectionInitializer = false)
```

Comment:
>BoundObjectInitializerMember [](start = 71, length = 28)

Do we ever expect to run into this node outside of the context of an object initializer? If not, then we should probably make new parameter required and simply remove the case handling this node in the switch at the top of this file. We should also make sure the right thing happens when someone requests IOperation node for the corresponding syntax directly, rather than getting to it through the parent.

Reply:
Yes, we'll see it for top-level initializers.

---
In reply to: [192269046](https://github.com/dotnet/roslyn/pull/27308#discussion_r192269046) [](ancestors = 192269046)